alf/algorithms/qrsac_algorithm.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2021 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Quantile Regression Soft Actor Critic Algorithm."""
+
+import torch
+import torch.distributions as td
+
+import alf
+from alf.algorithms.config import TrainerConfig
+from alf.algorithms.sac_algorithm import SacAlgorithm, SacInfo, ActionType
+from alf.algorithms.sac_algorithm import SacActionState, SacCriticState
+from alf.algorithms.sac_algorithm import SacCriticInfo, SacState
+from alf.data_structures import TimeStep
+from alf.data_structures import AlgStep
+from alf.nest import nest
+import alf.nest.utils as nest_utils
+from alf.networks import ActorDistributionNetwork, CriticNetwork
+from alf.networks import QNetwork
+from alf.tensor_specs import TensorSpec, BoundedTensorSpec
+from alf.utils import dist_utils
+
+
+def critic_network_cls(input_tensor_spec, quantile_size=64):
+    return alf.nn.Sequential(
+        alf.nn.EncodingNetwork(
```

Comment:
Seems like the conventionally used `CriticNetwork` has separate `EncodingNetwork` for the `observation` and `action` https://github.com/HorizonRobotics/alf/blob/de2c6a1a57e5572246bbddc75d9e1595787ac813/alf/networks/critic_networks.py#L132

Should we also follow the convention here, though we actually did not use such an network in the config file `alf/examples/qrsac_halfcheetah_conf.py`?

Reply:
I just want a simple default `critic_network_cls` here in case it is not specified in the conf file, I will replace it with a `QuantileCriticNetwork` as suggested by Wei.